proto/api/v2/report_service.proto

```diff
@@ -194,6 +207,28 @@ message DeleteReportRequest {
   string id = 1;
 }
 
+message ReportRequestViewBased {
+  enum ReportType {
+    VULNERABILITY = 0;
+  }
+  ReportType type = 1;
+  oneof filter {
+    ViewBasedVulnerabilityReportFilters view_based_vuln_report_filters = 2;
+  }
+
+  string area_of_concern = 3;
+  string query = 4;
+}
+
+message ReportHistoryResponseViewBased {
+  int32 report_snapshot_count = 1;
```

Comment:
nit: Do you need a count? It should be same as length of report_snapshots.

Reply:
We have it in design doc  but it is redundant and can be removed